server/src/main/java/io/deephaven/server/hierarchicaltable/HierarchicalTableViewSubscription.java

```diff
@@ -357,15 +360,16 @@ private static long buildAndSendSnapshot(
         // 5. Send the BarrageMessage
         try (final BarrageStreamGenerator streamGenerator =
                 streamGeneratorFactory.newGenerator(barrageMessage, writeMetricsConsumer)) {
-            // Note that we're always specifying "isInitialSnapshot=true". This is to provoke the subscription view to
-            // send the added rows on every snapshot, since (1) our added rows are flat, and thus cheap to send, and
-            // (2) we're relying on added rows to signal the full expanded size to the client.
-            GrpcUtil.safelyOnNext(listener,
-                    streamGenerator.getSubView(subscriptionOptions, true, true, rows, false, rows, rows, columns));
+            // initialSnapshot flag is ignored for non-growing viewports
+            final boolean initialSnapshot = false;
+            final boolean isFullSubscription = false;
+            GrpcUtil.safelyOnNext(listener, streamGenerator.getSubView(
+                    subscriptionOptions, initialSnapshot, isFullSubscription, rows, false,
+                    prevKeyspaceViewportRows, keyspaceViewportRows, columns));
         }
 
-        // 6. Let the caller know what the expanded size was
-        return expandedSize;
+        prevKeyspaceViewportRows.clear();
+        prevKeyspaceViewportRows.insert(keyspaceViewportRows);
```

Comment:
```suggestion
        prevKeyspaceViewportRows.resetTo(keyspaceViewportRows);
```

Reply:
As you noted, this probably needs to move into the try, because the `keyspaceViewportRows` is part of the `BarrageMessage`. That suggests (to me) that this aliasing may be error prone, and we should just have one name (`barrageMessage.rowsIncluded`) for the `keySpaceViewportRows`.